src/test/functional/pages/BuzzPage.ts

```diff
@@ -1,66 +1,194 @@
+        return this.page.getByRole('button', { name: Labels.SHARE, exact: true })
+      }
+
+    protected getFileInputButton(
+      ): Locator {
+        return this.page.locator('input[type="file"]')
+      }
+
+    protected getSharePhotoButton(
+    ): Locator {
+      return this.page.getByRole('button', { name: Labels.SHARE_PHOTOS })
+    }
+
+    public getCommentInput(
+      randomTitle: string
+    ): Locator {
+      return this.page
+        .locator('.oxd-sheet')
+        .filter({hasText:randomTitle})
+        .getByPlaceholder(Labels.WRITE_YOUR_COMMENT)
     }
 
-    public async locateElementWithDynamicText(randomTitle: string) {
-      const element = this.postBody.locator(`:has(:text('${randomTitle}'))`)
-      return element;
+    public getResharedPostButtonByRandomTitle(
+      randomTitle: string
+    ): Locator {
+      return this.page
+        .locator('.oxd-sheet')
+        .filter({hasText:randomTitle})
+        .locator('.bi-share-fill')
+    }
+
+    public getHeartButtonByRandomTitle(
+      randomTitle: string
+    ): Locator {
+      return this.page
+        .locator('.oxd-sheet')
+        .filter({hasText:randomTitle})
+        .locator('svg')
+    }
+
+    public getCommentPostButtonByRandomTitle(
+      randomTitle: string
+    ): Locator {
+      return this.page
+        .locator('.oxd-sheet')
+        .filter({hasText:randomTitle})
+        .locator('.bi-chat-text-fill')
+    }
+
+    public getPostWithRandomTitleAndPhoto(
+      randomTitle: string
+    ): Locator {
+      return this.page
+        .locator('.orangehrm-buzz-post-body')
+        .filter({hasText:randomTitle})
+        .filter({has: this.page.locator('.orangehrm-buzz-photos')})
+        .locator('.orangehrm-buzz-post-body-text')
+    }
+
+    // public getPostWithRandomTitleAndPhoto(randomTitle: string): Locator {
+    //   return this.page.locator(`.orangehrm-buzz-post-body:has(:text('${randomTitle}')):has(.orangehrm-buzz-photos) .orangehrm-buzz-post-body-text`);
+    // }
+    // this also works, let's keep it in codebase for poor times
+
+    public getOriginalTextOfReSharedPostWithPhoto(
+      randomTitle: string
+    ): Locator {
+      return this.page
+        .locator('.orangehrm-buzz-post-body')
+        .filter({hasText:randomTitle})
+        .filter({has: this.page.locator('.orangehrm-buzz-photos')})
+        .locator('.orangehrm-buzz-post-body-original-text')
+    }
+    
+    public getPostWithRandomTitleWithoutPhoto(
+      randomTitle: string
+    ): Locator {
+      return this.page
+        .locator('.orangehrm-buzz-post-body')
+        .filter({hasText:randomTitle})
+        .locator('.orangehrm-buzz-post-body-text')
     }
 
+    // public getPostWithRandomTitleWithoutPhoto(randomTitle: string): Locator {
+    //   return this.page.locator(`.orangehrm-buzz-post-body:has(:text("${randomTitle}")) .orangehrm-buzz-post-body-text`);
+    // }
+    // this also works, let's keep it in codebase for poor times
```

Comment:
What does poor times mean? 😄

Reply:
poor times means time when we do not know how to do something then we will have 2 soluation maybe in different code only 1 of them will work so i would keep it in code